src/components/progress-tracker/progress-tracker.style.js

```diff
@@ -0,0 +1,182 @@
+import { margin } from "styled-system";
+import baseTheme from "../../style/themes/base";
+import {
+  OUTER_TRACKER_LENGTH,
+  PROGRESS_TRACKER_SIZES,
+  PROGRESS_TRACKER_VARIANTS,
+} from "./progress-tracker.config";
+
+const StyledProgressTracker = styled.div`
+  ${margin}
+  text-align: center;
+  white-space: nowrap;
+
+  ${({ isVertical }) => css`
+    ${!isVertical &&
+    `
+      width: ${OUTER_TRACKER_LENGTH};
+    `}
+    ${isVertical &&
+    `
+      height: ${OUTER_TRACKER_LENGTH};
+      display: flex;
+    `}
+  `}
+`;
+
+const StyledProgressBar = styled.span`
+  ${({ direction, isVertical, size, theme }) => css`
+    display: flex;
+    position: relative;
+    background-color: ${theme.progressBar.background};
+
+    ${!isVertical &&
+    css`
+      overflow-x: hidden;
+      height: ${getHeight(size)};
+      width: 100%;
+    `}
+    ${isVertical &&
+    css`
+      overflow-y: hidden;
+      width: ${getHeight(size)};
+      height: 100%;
+
+      ${direction === "up" &&
+      `
+        align-items: flex-end;
+      `}
+    `}
+  `}
+`;
+
+const StyledValue = styled.span`
+  ${({ isMaxValue, theme }) => css`
+    ${isMaxValue &&
+    `
+      color: ${theme.text.placeholder};
+    `}
+    ${!isMaxValue &&
+    `
+      font-weight: bold;
+    `}
+  `}
+`;
+
+const StyledValuesLabel = styled.span`
+  text-align: start;
+  display: flex;
+  justify-content: space-between;
+  ${({ isVertical, position }) => css`
+    ${isVertical &&
+    css`
+      flex-direction: column;
+
+      ${position !== "left" &&
+      css`
+        padding-left: 4px;
+      `}
+
+      ${position === "left" &&
+      css`
+        padding-right: 4px;
+
+        ${StyledValue} {
+          text-align: right;
+        }
+      `}
+    `}
+
+    ${!isVertical &&
+    css`
+      ${position !== "bottom" &&
+      css`
+        padding-bottom: 4px;
+      `}
+
+      ${position === "bottom" &&
+      css`
+        padding-top: 4px;
+      `}
+    `}
+  `}
+`;
+
+const InnerBar = styled.span`
+  ${({ isVertical, progress, size, theme, variant }) => css`
+    position: absolute;
+    left: 0;
+    background-color: ${getInnerBarColour(variant, progress, theme)};
+
+    ${!isVertical &&
+    css`
+      width: calc(${OUTER_TRACKER_LENGTH} * ${progress / 100});
+      min-width: 2px;
+      height: ${getHeight(size)};
+    `}
+    ${isVertical &&
+    css`
+      height: calc(${OUTER_TRACKER_LENGTH} * ${progress / 100});
+      min-height: 2px;
+      width: ${getHeight(size)};
+    `}
+  `}
+`;
+
+function getHeight(size) {
+  switch (size) {
+    case "small":
+      return "4px";
+    case "large":
+      return "16px";
+    default:
+      return "8px";
+  }
+}
+
+function getInnerBarColour(variant, progress, theme) {
+  if (progress >= 100) return theme.colors.success;
+  if (variant === "default") return theme.progressBar.innerBackground;
+  if (progress < 20) return theme.colors.error;
+  return theme.colors.primary;
```

Comment:
@harpalsingh can you take a look at this as Sam's correct there's no mention of theme in the DS, I left that as is as I assumed Rafal had had a steer about it?

Reply:
@edleeks87 I agree, the design provided has no usage of theme, which for me is correct, this component is generic like forms, so it has a positive or negative value but then the "Slate" as the neutral color, so there should be no theme values added.

Reply:
Thanks @harpalsingh, for now I think it's best if we stick to what the DS decides, if the design updates at a later stage it will be easy enough to update